app/models/personal_access_token.rb

```diff
@@ -0,0 +1,62 @@
+class PersonalAccessToken < ActiveRecord::Base
+  include Authorizable
+  include Expirable
+
+  belongs_to :user
+
+  audited :except => [:token], :associated_with => :user
+
+  scoped_search :on => :name
+  scoped_search :on => :user_id, :complete_enabled => false, :only_explicit => true, :validator => ScopedSearch::Validators::INTEGER
+
+  validates_lengths_from_database
+
+  validates :token, :user_id, :token, presence: true
+  validates :name, presence: true, uniqueness: {scope: :user_id}
+
+  scope :active, -> { where(revoked: false).where("expires_at >= ? OR expires_at IS NULL", Time.current) }
+  scope :inactive, -> { where(revoked: true).or(where("expires_at < ?", Time.current)) }
+
+  attr_accessor :token_value
+
+  def self.authenticate_user(user, token)
+    tokens = self.active.where(:user => user, :token => encrypt_token(user, token))
+    return false unless tokens.any?
+    tokens.update_all(:last_used_at => Time.current, :updated_at => Time.current)
```

Comment:
could there be more than one active token?

Reply:
There should only be one token given that the tokes are unique.